### DISCUSSION TITLE
How to make physics deterministic?

### DISCUSSION THREAD

**Discussion by eigu47** [pmndrs/react-three-rapier#333] [Q&A]
Hello. I am trying to make a online pool game but i can't recreate the same shot with identical force and initial positions so both clients ends up desynchronizing.

Here i tried to recreate the situation. Click on white ball to shot with a fixed force and any other ball to reset positions.

https://codesandbox.io/embed/testing-pool-table-f03hdo

I don't understand if im doing something wrong or the engine is not deterministic.

**Comment by devjdeed** [comment]
Did you end up figuring out how to make each run similar
